### DISCUSSION TITLE
Variable dependent yield stress

### DISCUSSION THREAD

**Discussion by shipinqin** [idaholab/moose#16111] [Q&A Modules: Solid mechanics] [answered]
Hello,

I am trying to define concentration-dependent yield stress in MOOSE, but not sure how to do it (more details on what I am trying to do are below).  Any suggestions from you will be appreciated.  

The concentration (actually, it is the fraction of a phase) is read from a MOOSE simulation result file generated in another MOOSE simulation, and I am trying to make the yield stress to be dependent on this concentration.  As a starting point, I worked on the simplest strain hardening case - J2 with perfect plasticity, so there is only one parameter, the initial yield stress, that needs to be dependent on the concentration (but I do not know how).

As of now, I have successfully made the Young's modulus to be dependent on the concentration by using the "**ComputeVariableIsotropicElasticityTensor**" app.  However, there does not seem to be a yield stress counterpart.  Currently, I implemented the J2 plasticity model through "**IsotropicPlasticityStressUpdate**" and "**ComputeMultipleInelasticStress**", however, "_yield_stress_" in "**IsotropicPlasticityStressUpdate**" takes only numbers, not variables.

Again, any suggestions will be appreciated!

Thanks.

**Comment by bwspenc** [comment, marked as answer]
You would need to change `yield_stress` to be a coupled variable. Look at `ComputeSmearedCrackingStress`, where `cracking_stress` was recently switched to take a variable in a similar way. If it's constant, the user can supply a single number as before for that parameter. You do need to give some thought to how a variable yield stress would affect the hardening behavior, though.

**Reply by shipinqin** [reply]
Hello Ben,

Thank you very much for your suggestions.  I made it work today following them!  

Like you have said, the suggested workaround is ok when assuming perfect plasticity, however, things will be more challenging if strain hardening is involved.  While I am not rushed to move to that stage for now, I am indeed interested to know what you think is the best possible solution.  Is modifying the IsotropicPlasticityStressUpdate.C file a good idea?  I tried to change `params.addParam<Real>` in line 28 to `params.addParam<MaterialPropertyName>` and `getParam<Real>` in line 49 to `getMaterialProperty<Real>`, but apparently it did not work (not surprising).  Will modifications like this be trivial, or it will actually involve significant modifications to many files?

Thanks.
